Maple2.Tools/Collision/BoundingBox.cs

```diff
@@ -23,7 +23,9 @@ public BoundingBox(in Vector2 vector1, in Vector2 vector2) {
     }
 
     public override bool Contains(in Vector2 point) {
-        return min.X <= point.X && point.X <= max.X && min.Y <= point.Y && point.Y <= max.Y;
+        int pointX = (int) point.X;
+        int pointY = (int) point.Y;
+        return min.X <= pointX && pointX <= max.X && min.Y <= pointY && pointY <= max.Y;
```

Comment:
Should be using an epsilon argument for float Vector2 types, not converting to ints. If int bounds are needed then an integer Vector2 type should be used instead.

```suggestion
    public override bool Contains(in Vector2 point, float epsilon = 1e-5f) {
        return min.X - epsilon <= pointX && pointX <= max.X + epsilon && min.Y - epsilon <= pointY && pointY <= max.Y + epsilon;
    }
```